packages/utils/src/types.ts

```diff
@@ -3,7 +3,168 @@ import {
   SnapKeyring as Keyring,
 } from '@metamask/snap-types';
 import { Json } from '@metamask/utils';
-import { NpmSnapPackageJson, SnapManifest } from './json-schemas';
+import {
+  any,
+  boolean,
+  Infer,
+  is,
+  literal,
+  object,
+  optional,
+  pattern,
+  record,
+  refine,
+  size,
+  string,
+  type,
+  union,
+} from 'superstruct';
+import { valid } from 'semver';
+import { minSize } from './structs';
+import { assertStruct } from './assert';
+
+export enum NpmSnapFileNames {
+  PackageJson = 'package.json',
+  Manifest = 'snap.manifest.json',
+}
+
+/**
+ * A struct for validating a version string.
+ */
+export const VersionStruct = refine(string(), 'Version', (value) => {
+  return valid(value) !== null;
+});
+
+export const NameStruct = size(
+  pattern(
+    string(),
+    /^(?:@[a-z0-9-*~][a-z0-9-*._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/u,
+  ),
+  1,
+  214,
+);
+
+// Note we use `type` instead of `object` here, because the latter does not
+// allow unknown keys.
+export const NpmSnapPackageJsonStruct = type({
+  version: VersionStruct,
+  name: NameStruct,
+  private: optional(boolean()),
```

Comment:
I'm pretty sure we don't use some of those fields

Reply:
Removed them. We can add them again in the future if necessary.